mpmath/libmp/libintmath.py

```diff
@@ -297,6 +297,12 @@ def sqrtrem_python(x):
 
 def isqrt_python(x):
     """Integer square root with correct (floor) rounding."""
+    # Due to an implementation problem with `math.isqrt`,
+    # the speed is reversed around about 2**55000.
+    # Python 3.12 will resolve this, so we assume that in
+    # the future we will just call `math.isqrt`.
+    if x.bit_length() < 55000:
+        return math.isqrt(x)
     return sqrtrem_python(x)[0]
```

Comment:
Maybe this whole thing could be wrapped in a Python version check:
```python
if sys.version_info >= (3, 12):
   def isqrt_python(...)
else:
   def isqrt_python(...)
```
I wonder what pypy does to provide isqrt.

Reply:
> wrapped in a Python version check

Yep.  I would also prefer this.

> I wonder what pypy does to provide isqrt.

https://foss.heptapod.net/pypy/pypy/-/blob/branch/py3.10/pypy/module/math/app_math.py#L133

Reply:
```python
else:
    if sys.version_info >= (3, 12):
        isqrt_small = isqrt_fast = isqrt = math.isqrt
    else:
        isqrt_small = isqrt_small_python
        isqrt_fast = isqrt_fast_python
        isqrt = isqrt_python
```
Yes.

Reply:
Fixed.